contracts/ERC20Safe.sol

```diff
@@ -132,14 +166,40 @@ contract ERC20Safe is BridgeRole, Pausable {
       @notice Entrypoint for the user in the bridge. Will create a new deposit
       @param tokenAddress Address of the contract for the ERC20 token that will be deposited
       @param amount number of tokens that need to be deposited
-      @param recipientAddress address of the receiver of tokens on Elrond Network
+      @param recipientAddress address of the receiver of tokens on MultiversX Network
       @notice emits {ERC20Deposited} event
 \   */
-    function deposit(
-        address tokenAddress,
-        uint256 amount,
-        bytes32 recipientAddress
-    ) public whenNotPaused {
+    function deposit(address tokenAddress, uint256 amount, bytes32 recipientAddress) public whenNotPaused {
+        uint112 batchNonce;
+        uint112 depositNonce;
+        (batchNonce, depositNonce) = _deposit_common(tokenAddress, amount, recipientAddress);
+        emit ERC20Deposit(depositNonce, batchNonce);
+    }
+
+    /*
+     * @notice Entrypoint for the user in the bridge. Will create a new deposit
+     * @dev The `callData` parameter is structured to include the endpoint name, gas limit, and arguments for the cross-chain call.
+     *
+     * @param tokenAddress The address of the ERC20 token to deposit.
+     * @param amount The amount of tokens to deposit.
+     * @param recipientAddress The address on the target chain to receive the tokens.
+     * @param callData The encoded data specifying the cross-chain call details. The expected format is:
+     *        0x + endpoint_name_length (4 bytes) + endpoint_name + gas_limit (8 bytes) +
+     *        01 (ArgumentsPresentProtocolMarker) + num_arguments_length (4 bytes) + [argument_length (4 bytes) + argument]...
+     *        This payload includes the endpoint name, gas limit for the execution, and the arguments for the call.
+     *        In case of no arguments, only the ArgumentsMissingProtocolMarker should be included. The expected format is:
+     *        0x + endpoint_name_length (4 bytes) + endpoint_name + gas_limit (8 bytes) +
+     *        00 (ArgumentsPresentProtocolMarker)
```

Comment:
Since this is completely unchecked and not executed at this smart contract level, the complete description should be present on the other side of the bridge, where the value is read and used.

This will give you the freedom to change the "exit", without also updating the input comments of `callData`.

Reply:
I would still keep it here as most likely the solidity developers will not read the SpaceVM contracts, This is actually the encoding implementation on the MultiversX and it will not change

Reply:
Sounds good